lib/kafka/protocol/message_set.rb

```diff
@@ -3,9 +3,14 @@ module Protocol
     class MessageSet
       attr_reader :messages
 
-      def initialize(messages: [], compression_codec: nil)
+      def initialize(messages: [], compression_codec: nil, compression_threshold: 1)
```

Comment:
> I assume you want to have a null codec that simply doesn't touch the messages?

That’s exactly what I had in mind. But now I see the differences in `encode_with_compression` and `encode_without_compression `.